autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/HypervisorManagerTest.java

```diff
@@ -0,0 +1,121 @@
+ * an open source autonomic cloud computing management platform.
+ * Copyright (C) 2016 Autonomiccs, Inc.
+ *
+ * Licensed to the Autonomiccs, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The Autonomiccs, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package br.com.autonomiccs.autonomic.administration.plugin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.cloud.host.Host;
+import com.cloud.host.HostVO;
+import com.cloud.hypervisor.Hypervisor.HypervisorType;
+import com.cloud.utils.exception.CloudRuntimeException;
+
+import br.com.autonomiccs.autonomic.administration.plugin.hypervisors.HypervisorHost;
+import br.com.autonomiccs.autonomic.administration.plugin.hypervisors.xenserver.XenHypervisor;
+import br.com.autonomiccs.autonomic.plugin.common.services.HostService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HypervisorManagerTest {
+
+    @Spy
+    @InjectMocks
+    private HypervisorManager hypervisorManager;
+    @Mock
+    private HostService hostService;
+    @Mock
+    private List<HypervisorHost> hypervisorHosts;
+
+    private HypervisorHost hypervisor;
+
+    @Before
+    public void setup() {
+        hypervisor = Mockito.mock(XenHypervisor.class);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shutdownHostTestFull() {
+        HostVO host = setTestOfShutdownHost(HypervisorType.XenServer, Host.Type.Routing, true);
+        hypervisorManager.shutdownHost(host);
+        verifyShutdownHostCalledMethods(1, 1);
+    }
+
+    @Test
+    public void shutdownHostTestNotRoutingHost() {
+        HostVO host = setTestOfShutdownHost(HypervisorType.XenServer, Host.Type.ConsoleProxy, true);
+        hypervisorManager.shutdownHost(host);
+        verifyShutdownHostCalledMethods(0, 0);
+    }
+
+    @Test(expected = CloudRuntimeException.class)
+    public void shutdownHostTestNotXenServer() {
+        HostVO host = setTestOfShutdownHost(HypervisorType.XenServer, Host.Type.Routing, false);
```

Comment:
The name of this test implies that the host is not running a xenserver hypervisor yet the test is setup using xenserver as the hypervisorType? Am I missing something?

On that subject, is there a way for us to have a test case that is set up with a generic hypervisor that will never be supported so that it will never trigger that condition ( `if (currentHypervisorHost.supportsHypervisor(hostVo.getHypervisorType()))` ), since future plans include support for more hypervisors?